airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/SchemaViewer/useGetSourceSchema.ts

```diff
@@ -0,0 +1,41 @@
+import { JSONSchema4 } from "json-schema";
+import { useQuery } from "react-query";
+
+import { useConfig } from "packages/cloud/services/config";
+
+const fetchSchema = async ({
+  apiUrl,
+  sourceDefinitionId,
+}: {
+  apiUrl: string;
+  sourceDefinitionId: string;
+}): Promise<JSONSchema4> => {
+  const path = `${apiUrl}/source_definition/${sourceDefinitionId}/discover_schema`;
+
+  const response = await fetch(path);
+
+  if (!response.ok) {
+    throw new Error("Schema not found");
+  }
+  return response.json();
+};
```

Comment:
I'm not sure if this is the best approach. I'm throwing an error here because it seems that, when using `fetch` with `react-query` getting a 404 doesn't set `{status: 'error'}`  on `useQuery` response.

Happy to see if there are cleaner ways to do this